boards: shields: x_nucleo_wb05kn1: add support for b_u585i_iot02a board

whati001 opened the pull request:
Ported x_nucleo_wb05kn1 shield for st b_u585i_iot02a board

erwango reviewed (commented):
I agree this is correct but I'm not sure it makes sense to push upstream.
You might need it on your end for experimentation but adding a BLE shield on a board with an existing BLE module might not be a case worst sharing.

Commit:
boards: shields: x_nucleo_wb05kn1: add support for b_u585i_iot02a board

Add b_u585i_iot02a overlay file for x_nucleo_wb05kn1 shield to
enhance usability.

Signed-off-by: Andreas Karner <andreas.karner@sloc.one>
